compiler/compile/VirtualGuard.cpp

```diff
@@ -213,6 +213,60 @@ TR_VirtualGuard::createMethodGuard
    return createMethodGuardWithReceiver (kind, comp, calleeIndex, callNode, destination, calleeSymbol, thisClass, callNode->getSecondChild());
    }
 
+/*
+ * Shape of a breakpoint guard:
+ * iflcmpeq
+ *    land
+ *       aloadi <ConstantPool shadow>
+ *          aconst J9Method
+ *       isBreakpointedBit 
+ *    isBreakpointedBit 
+ */
+TR::Node*
+TR_VirtualGuard::createBreakpointGuard
+(TR::Compilation * comp, int16_t calleeIndex,
+ TR::Node* callNode, TR::TreeTop * destination, TR::ResolvedMethodSymbol * calleeSymbol)
+   {
+#ifdef J9_PROJECT_SPECIFIC
```

Comment:
The only part of this that is J9 Project specific is the actual test. If we have a convention about the test and have a method that generates the test trees the rest of the code is OMR common and doesn't need to be hidden - the Java specific part could then move out of OMR to leave a common language framework that just requires implementing languages to generate the trees to do the test for the breakpoint. If that is a huge amount of work then could we at least isolate the J9 Project specific part in a function that describes what it is generating and for non Java languages asserts or something similar?